source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -855,6 +907,260 @@ namespace AdaptiveCards::Rendering::Uwp::ActionHelpers
+    }
+
+    HRESULT BuildInlineShowCard(_In_opt_ ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveCard* adaptiveCard,
+                                _In_opt_ IAdaptiveActionSet* adaptiveActionSet,
+                                _In_ IAdaptiveActionElement* showCardAction,
+                                _In_ IPanel* showCardsPanel,
+                                _In_ IUIElement* actionButton,
+                                _In_ IUIElement* actionOverflowItem,
+                                UINT buttonIndex,
+                                _In_ IAdaptiveRenderContext* renderContext,
+                                _In_ IAdaptiveRenderArgs* renderArgs)
+    {
+        ComPtr<IAdaptiveActionElement> action(showCardAction);
+
+        // Check the host config to see if we need to build inline show cards
+        ComPtr<IAdaptiveHostConfig> hostConfig;
+        RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+        ComPtr<IAdaptiveActionsConfig> actionsConfig;
+        RETURN_IF_FAILED(hostConfig->get_Actions(actionsConfig.GetAddressOf()));
+
+        ComPtr<IAdaptiveShowCardActionConfig> showCardActionConfig;
+        RETURN_IF_FAILED(actionsConfig->get_ShowCard(&showCardActionConfig));
+
+        ABI::AdaptiveCards::Rendering::Uwp::ActionMode showCardActionMode;
+        RETURN_IF_FAILED(showCardActionConfig->get_ActionMode(&showCardActionMode));
+
+        if (showCardActionMode == ABI::AdaptiveCards::Rendering::Uwp::ActionMode::Inline)
+        {
+            // Get the card to be shown
+            ComPtr<IAdaptiveShowCardAction> actionAsShowCardAction;
+            RETURN_IF_FAILED(action.As(&actionAsShowCardAction));
+
+            ComPtr<IAdaptiveCard> showCard;
+            RETURN_IF_FAILED(actionAsShowCardAction->get_Card(&showCard));
+
+            // Render the card and add it to the show card panel
+            ComPtr<IUIElement> uiShowCard;
+            RETURN_IF_FAILED(AdaptiveShowCardActionRenderer::BuildShowCard(
+                showCard.Get(), renderContext, renderArgs, (adaptiveActionSet == nullptr), uiShowCard.GetAddressOf()));
+
+            XamlHelpers::AppendXamlElementToPanel(uiShowCard.Get(), showCardsPanel);
+
+            // Register the show card with the render context
+            ComPtr<AdaptiveCards::Rendering::Uwp::AdaptiveRenderContext> contextImpl =
+                PeekInnards<AdaptiveCards::Rendering::Uwp::AdaptiveRenderContext>(renderContext);
+
+            if (adaptiveActionSet)
+            {
+                RETURN_IF_FAILED(contextImpl->AddInlineShowCard(
+                    adaptiveActionSet, actionAsShowCardAction.Get(), actionButton, actionOverflowItem, uiShowCard.Get(), buttonIndex, renderArgs));
+            }
+            else
+            {
+                RETURN_IF_FAILED(contextImpl->AddInlineShowCard(
+                    adaptiveCard, actionAsShowCardAction.Get(), actionButton, actionOverflowItem, uiShowCard.Get(), buttonIndex, renderArgs));
+            }
```

Comment:
It seems like the compiler can't deal with this since the switch is between two different overloads:
> ':': no conversion from 'ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveCard *' to 'ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveActionSet *'

Reply:
hmm... -- that's unfortunate :(